pandas/core/arrays/categorical.py

```diff
@@ -655,13 +655,29 @@ def from_codes(cls, codes, categories, ordered=False):
         codes : array-like, integers
             An integer array, where each integer points to a category in
             categories or -1 for NaN
-        categories : index-like
+        categories : index-like, optional
             The categories for the categorical. Items need to be unique.
-        ordered : boolean, (default False)
-            Whether or not this categorical is treated as a ordered
+        ordered : bool, optional
+            Whether or not this categorical is treated as an ordered
             categorical. If not given, the resulting categorical will be
             unordered.
+
+            .. versionchanged:: 0.24.0
+
+                The default value has been changed to  ``None``. Previously
+                the default value was ``False``.
+        dtype : CategoricalDtype, optional
+            An instance of ``CategoricalDtype`` to use for this categorical.
+
+            .. versionadded:: 0.24.0
         """
+        if dtype is not None:
+            if categories is not None or ordered is not None:
+                raise ValueError("Cannot specify both `dtype` and `categories`"
+                                 " or `ordered`.")
```

Comment:
* Can you add that this can be raised in the docs?
* The error message confuses me.  Are you saying: both "dtype" and ("categories" / "ordered") ?

Reply:
I copied that message from Categorical.__init__, but fair enough, I've changed it in both locations.

Reply:
Fantastic.  Thanks for doing that!